test-data/unit/stubgen.test

```diff
@@ -510,19 +560,19 @@ class A:
 [out]
 class A:
     class B:
-        x = ...  # type: int
+        x: int
         def f(self): ...
     def g(self): ...
 
 [case testExportViaRelativeImport]
 from .api import get
 [out]
-from .api import get as get
```

Comment:
Even after reading this comment and pep-484, I'm not sure I fully understand what is required here. What I have is:
 * If a stub needs to export an imported name, it should import it with an alias (even if the alias is name is the same as the original). Is that correct?
 * But then, how does stubgen know which imported names where "intended" to be reexported? should it use `__all__` and ignore everything else? should it reexport everything just in case? should it reexport only names imported in a certain way?

I could use a hand here at least with the test cases, once I have some boundaries defined by those I could probably tweak the implementation to match the spec.

Reply:
> If a stub needs to export an imported name, it should import it with an alias (even if the alias is name is the same as the original). Is that correct?

Yes.
> But then, how does stubgen know which imported names where "intended" to be reexported? should it use __all__ and ignore everything else? should it reexport everything just in case? should it reexport only names imported in a certain way?

I think the safest bet would be to re-export all globally visible names (i.e. ignore imports in function bodies), if there is not ``__all__`` defined, _and_ only re-export names form ``__all__`` if there is one.

Reply:
I'm implementing the `__all__` scenario. I'm not doing the "reexport everything if there's no `__all__`" because I think the result ends up being unnecessarily noisy for most cases (you'll end up with tons of modules saying `from typing import List as List` because they don't have an `__all__`). In any case, it's easy to change our minds later

Reply:
> you'll end up with tons of modules saying ``from typing import List as List`` because they don't have an ``__all__``.

You can just manually exclude names imported from ``typing``. In general, this is an important point, since it already caused some bugs in typeshed, see e.g. https://github.com/python/typeshed/pull/1484. I would prefer this test to work.

(Also I don't think people will be unhappy about a bit more bulky stub, but they will be certainly unhappy about false positives when ``as name`` isn't present).